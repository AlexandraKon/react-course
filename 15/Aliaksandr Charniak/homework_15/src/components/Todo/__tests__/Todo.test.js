import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

import Todo from '..';

configure({ adapter: new Adapter() });

describe('<Todo /> mounting behaviour', () => {
  const mockFn = jest.fn();

  const props = {
    onClick: mockFn,
    completed: false,
    text: 'Some text...',
  };

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Todo {...props} />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render 1 <Todo /> component', () => {
    expect(wrapper).toHaveLength(1);
  });

  it('should correctly render children', () => {
    // console.log(wrapper.props());
    expect(wrapper.props().children).toEqual(props.text);
  });

  it('set props correctly', () => {
    const newText = 'Hello!';
    wrapper.setProps({ text: newText });
    expect(wrapper.props().children).toEqual(newText);
  });

  it('should call onClick handler when clicked', () => {
    wrapper.simulate('click');
    expect(mockFn).toHaveBeenCalledTimes(1);
  });
});

describe('<Todo /> styling behaviour', () => {
  let wrapper;
  const mockFn = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <Todo onClick={mockFn} completed={false} text='Some text...' />
    );
  });

  it('should not path linethrow style, when todo is incomplete', () => {
    expect(wrapper.props().style).toEqual({ textDecoration: 'none' });
  });

  it('should path linethrow style, when todo is complete', () => {
    wrapper.setProps({ completed: true });

    expect(wrapper.props().style).toEqual({ textDecoration: 'line-through' });
  });
});
