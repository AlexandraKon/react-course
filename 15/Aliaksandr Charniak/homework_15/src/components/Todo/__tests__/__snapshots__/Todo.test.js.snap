// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Todo /> mounting behaviour should match snapshot 1`] = `
<li
  onClick={[MockFunction]}
  style={
    Object {
      "textDecoration": "none",
    }
  }
>
  Some text...
</li>
`;
